### PR TITLE
Add new "Getting Started with Python" doc.

### PR DESCRIPTION
**Pull request by ScottTodd**
Plenty of room for improvement (complete and document CMake support, publish pip package(s), share Colab notebooks that don't require building from source, etc.), but this lets us cut away from the old documentation.

Also in this PR:

* Removed old docs,  now that all of their content is covered in the new docs
* Tweaked some website styling
* Fixed an encoding issue on Windows in `update_op_coverage.py`

Fixes https://github.com/google/iree/issues/1363.

### CODE REVIEW

File: docs/GetStarted/getting_started_python.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Python bindings you must build from source.
+
+## Prerequisites
+
+You should already have IREE cloned and building on your machine. See the other
+[getting started guides](.) for instructions.
+
+> Note:<br>
+> &nbsp;&nbsp;&nbsp;&nbsp;Support is best with Bazel.
+> For CMake (excluding TensorFlow), set the `IREE_BUILD_PYTHON_BINDINGS` option.
```
Comment (ScottTodd) [inline docs/GetStarted/getting_started_python.md:17]: fyi @marbre 

could document CMake Python support in here (particularly how to run tests on `bindings/python/...`?)

File: scripts/update_op_coverage.py [hunk 2]
```diff
@@ -124,6 +125,6 @@ def generate_table(build_dir):
   args = parse_arguments()
   content = generate_table(args.build_dir)
   table_path = os.path.join(args.build_dir, 'doc', 'op_coverage.md')
-  with open(table_path, 'w') as f:
+  with open(table_path, 'w', encoding='utf-8') as f:
```
Comment (ScottTodd) [inline scripts/update_op_coverage.py:128]: fyi @hanhanW, needed this on Windows

Also see the "Note" callout format used above. Trying to use similar styling throughout the published docs. (I'm not totally happy with how they look though, [here](https://google.github.io/iree/GetStarted/LinuxCMake#install-cmake) is a sample).

Comment (hanhanW) [reply scripts/update_op_coverage.py:128]: Thanks for fixing this, I missed the encoding issue..

File: docs/GetStarted/getting_started_python.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Python Setup
+
+Install a recent version of [Python 3](https://www.python.org/downloads/) and
+[pip](https://pip.pypa.io/en/stable/installing/), if needed.
+
+Install packages:
+
+```shell
+$ pip install numpy
```
Comment (stellaraccident) [inline docs/GetStarted/getting_started_python.md:27]: Sadly, this often produces bad results.

For canonical documentation, I'd recommend: `python3 -m pip install ...`

Why? In multi-python-version regimes (pretty much everything these days), almost the only thing you can easily inspet/control is which interpreter you are invoking. This form makes it explicit.

File: docs/GetStarted/getting_started_python.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+To run tests for the TensorFlow integration, which include end-to-end backend
+comparison tests:
+
+```shell
+# Exclude tests that are skipped in the GitHub Actions ("ga") CI
+$ bazel test \
+  --build_tag_filters="noga" \
+  --test_tag_filters="noga" \
+  --define=iree_tensorflow=true \
```
Comment (stellaraccident) [inline docs/GetStarted/getting_started_python.md:49]: I'm not sure we need this anymore. Also not 100% sure we don't.

Comment (ScottTodd) [reply docs/GetStarted/getting_started_python.md:50]: Still used here: https://github.com/google/iree/blob/b2c034f78f0eb3f85cb50ff48ba594c4aa12b5aa/colab/BUILD.bazel#L27-L33

File: docs/GetStarted/getting_started_python.md (hunk 3 above)
Comment (ScottTodd) [reply docs/GetStarted/getting_started_python.md]: Oh right, thanks for catching that. I've updated to that format and added `python3 -m pip install --upgrade pip`, which we found necessary for `tf-nightly`.

**Pull request closed without merging**